test/e2e/utils/test_context.go

```diff
@@ -54,30 +50,36 @@ func NewTestContext(binaryName string, env ...string) (*TestContext, error) {
 		return nil, err
 	}
 
-	testGroup := "bar" + testSuffix
-	path, err := filepath.Abs("e2e-" + testSuffix)
+	cc := &CmdContext{
+		Env: env,
+	}
+
+	// Use kubectl to get Kubernetes client and cluster version.
+	kubectl := &Kubectl{
+		Namespace:  fmt.Sprintf("e2e-%s-system", testSuffix),
+		CmdContext: cc,
+	}
+	k8sVersion, err := kubectl.Version()
 	if err != nil {
 		return nil, err
 	}
 
-	cc := &CmdContext{
-		Env: env,
-		Dir: path,
+	// Set CmdContext.Dir after running Kubectl.Version() because dir does not exist yet.
+	if cc.Dir, err = filepath.Abs("e2e-" + testSuffix); err != nil {
+		return nil, err
 	}
 
 	return &TestContext{
 		TestSuffix: testSuffix,
 		Domain:     "example.com" + testSuffix,
-		Group:      testGroup,
+		Group:      "bar" + testSuffix,
```

Comment:
```suggestion
		Group:      testSuffix,
```
why we would add "bar"?

Reply:
The group setup hasn't been changed, look at the old group setup.

Reply:
Catcher line 57 `testGroup := "bar" + testSuffix`